### ISSUE TITLE
Portalled Select.Positioner has `aria-hidden="true"` when open inside a Dialog

### ISSUE THREAD

**Issue by StyleShit** [chakra-ui/ark#3728]
### Description

When rendering a Portalled `Select.Positioner` component inside a `Dialog`, it has an `aria-hidden="true"` prop which causes screen readers not to show it (and its descendants) in the a11y tree.

### Link to Reproduction (or Detailed Explanation)

https://stackblitz.com/edit/occkdfxt?file=src%2FApp.tsx

### Steps to Reproduce

1. Open the select which is outside the dialog
2. Check its positioner DOM node
3. You'll see it doesn't have any `aria-hidden` prop
4. Now open the select which is inside the dialog
5. Check its positioner DOM node
6. You'll see that is has an `aria-hidden="true"` prop
7. Removing the `Portal` from the `Select` component will fix the issue

### Ark UI Version

Latest (5.29.1)

### Framework

- [x] React
- [ ] Solid
- [ ] Vue
- [ ] Svelte

### Browser

Firefox 145.0b9, Chrome 142.0.7444.176

**Comment by segunadebayo** [comment]
Consider lazy-mounting the select so it's not rendered in the DOM until opened.

Pass the `lazyMount` prop to the `Select.Root`
